### DISCUSSION TITLE
Flutter Desktop App Features Suggestion

### DISCUSSION THREAD

**Discussion by ghost** [ente-io/ente#1022] [Enhancements]
**Is your feature request related to a problem? Please describe.**
Not really a problem since the app is in beta officially

**Describe the solution you'd like**
Here are some features I am suggesting to implement in the future:
~ [Flutter System Tray](https://github.com/leanflutter/tray_manager) (Left Click to open the app in the tray cursor location)
~ Close application in tray (setting that could be changed)
~ Local Passcode (with inactivity timeout that re-locks & encrypts the data based of inactivity ; different from the account password by itself)
~ Add to startup 
~ Remember window size between starts (Will be most likely implemented in the near future, but just a heads up)

Also, thank you guys for this amazing auth application; I just got redirected here after the announcement of Authy EOL and I was kind of mad that there was no great alternative, until today ;)

**Comment by prateekmedia** [comment]
The Local Passcode is not implemented other than that all will be available in stable version.
Note that if you want to add to startup then you will have to select when installing on windows or you can manually add based on your operating system startup apps, we haven't provided a dedicated settings for it as such.

**Reply by gitmeED331** [reply]
I have the open on startup issue solved/handled, but what is missing is the ability "start-in-tray" and "close-to-tray", which are much needed, especially for those on non-traditional desktop environments, like window compositors.

The current state of the desktop apps is a just one giant security risk that isn't being addressed.

OS: Arch Linux
Desktop: Hyprland
Shell: AGS

**Reply by rom4ster** [reply]
I could not agree more. Yes there is a tray icon but it is functionally useless. It does nothing except show up. What good is saying something is present when it is just for show.

![image](https://github.com/user-attachments/assets/8e389890-1df3-48fd-81a3-acce3a5cac3b)

As you can see in this option list, all this does is minimize, open window, or close the app.  To do ANY of these actions takes less clicks by just using the x [ ] - buttons or simply hitting the app on the taskbar. 

In fact, it is literally easier to right click the taskbar icon for closing the app or opening the window. 


There is no use case for the tray icon but the problem is *there should be*  

The reason people want a tray icon is so the app does not have to show up and clutter the taskbar. 

At the very least, the app should close to the tray. and then for a true quit, have the tray icon do it. But even that is just bare minimum. Ideally the tray icon should be able to show the remaining time for a code and give the option to (without showing the code) copy it. 

As it stands right now, the desktop app is just missing too many features. Its almost as if I am using an android app in an emulator and I think that it is being mostly ignored. I think its time to show it some love.

**Comment by gitmeED331** [comment]
The PIN is nice, thanks... But we shouldn't have to enter it to hide the window.

Some other issues that go along with this:
- double clicking on the icon in the tray does not open it.
- still no start-in-tray (I want it to start directly to the tray, not have to do it manually every time and I will open and unlock when I need to use it)
- there needs to be auto-lock functionality:
- - on timeout
- - on session-lock (based on system signals)
- the PIN length should be user specified
- the still no close-to-tray
